UI/audio/feature/(not so) minor visual jukebox changes

srbackman opened the pull request:
### Battle audio changes
- Added support for better battle sfx playback
- Cleaned up battle common sounds in SFXDataReference.

### Audio file changes
- Moved all remaining audio files from MenuUi to under Altzone\Audio.
- Removed some obsolete sounds and duplicate music tracks.

### Jukebox changes
- Added SwipeBlocker's to jukebox.
- Music track info popup.
- User now sees information over the track disk image based on what condition it is currently.
- A small brief text popup will inform the user when a track is successfully been added or removed.
- Improved several technical aspects of the jukebox.
- Created ArtistReference scriptable object for easier storage and usage of including artists in any place where it is wanted to give any credits for a artist or artists.

srbackman committed:
Jukebox music track info popup

- Added info popup for jukebox.

srbackman committed:
Merge branch 'main' into UI/Audio/feature/MinorVisualJukeboxChanges

srbackman committed:
Artist data & more work for music info popup

- Created ArtistReference scriptable object for easier storage and usage of including artists in any place where it is wanted to give any credits for a artist or artists.
- MusicReference now has an enum that can be used when getting a wanted category.
- Added JukeboxInfoPopupHandler.cs to handle the jukebox info popup.
- Added some visual updates to jukebox (some of them awaiting functionality).

srbackman committed:
Jukebox info popup partially functional

- Jukebox info popup now works partially
- Missing features: Set as favorite, add to playlist, preview from info popup & preview indicator on jukebox.

srbackman committed:
More jukebox information for user!

- Jukebox track info popup is now ready (except for the how many favorites count is not implemented).
- User now sees information over the track disk image based on what condition it is currently.
- A small brief text popup will inform the user when a track is successfully been added or removed.
- Improved several technical aspects of the jukebox.

srbackman committed:
Merge branch 'main' into UI/Audio/feature/MinorVisualJukeboxChanges

srbackman committed:
Merge branch 'main' into UI/Audio/feature/MinorVisualJukeboxChanges

srbackman committed:
Fixed merge conflict + jukebox updates

- Added missing DailyTask progress listener to JukeboxSongButton prefab due to a merge conflict.
- Added SwipeBlocker's to jukebox.
- Raised the PopUpJukebox2.png file texture size from 256 to 512 so it's less of a blur due to its use case size.

srbackman committed:
Moved and removed audio files

- Moved all remaining audio files from MenuUi to under Altzone\Audio.
- Removed some obsolete sounds and duplicate music tracks.
- Cleaned up battle common sounds in SFXDataReference.

srbackman committed:
Added support for better battle sfx playback

srbackman committed:
Merge branch 'main' into UI/Audio/feature/MinorVisualJukeboxChanges

BillTheBeast commented on Assets/Altzone/Scripts/Reference Sheets/ArtistReference.cs:
```diff
@@ -0,0 +1,13 @@
+    [CreateAssetMenu(fileName = "ArtistReference", menuName = "ScriptableObjects/ArtistReferenceScriptableObject")]
+    public class ArtistReference : ScriptableObject
+    {
+        public string Name;
```
Itse varmaan tekisin näistä read onlyn Getterin avulla, koska tuskin tarvitsee näitä muuttaa kun peli on käynnissä.

BillTheBeast commented on Assets/MenuUi/Scripts/Jukebox/JukeboxTextPopup.cs:
```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using UnityEngine;
+
+public class JukeboxTextPopup : MonoBehaviour
```
Etkö pystyisi käyttämään PopupController skriptiä/InfoPopUp prefabiä tätä varten? (Sitä joutuu ehkä vähän siistimään)

srbackman committed:
Replaced JukeboxTextPopup with PopupController

- Replaced JukeboxTextPopup with PopupController.
- Made getters for ArtistReference.

BillTheBeast commented:
Jukeboxin esikuuntelu päättyy ensiaikaisesti, jos jukeboxi suorittaa serverikutsun hakeakseen ajantasaisen biisilistan.

BillTheBeast commented:
Jukeboxin biisien scrollaaminen aiheuttaa helposti virhepainalluksia, koska se tunnistaa siitä aiheutuvat painallukset myös valinta painalluksina.

BillTheBeast commented on Assets/Altzone/Scripts/Audio/SFXHandler.cs:
```diff
@@ -95,6 +96,55 @@ private float GetVolume(SoundEffect soundEffect)
+            return _activeChannels[_activeChannels.Count - 1];
+        }
+
+        public ActiveChannelPath? Play(AudioCategoryType categoryType, BattleSFXNameTypes battleSFXName, string mainMenuMusicName, float pitch)
```
Jäikö tuo mainMenuMusicName tuohon valingossa kun kopioit tuota metodia ylläolevasta?

BillTheBeast commented on Assets/MenuUi/Scripts/Jukebox/JukeboxInfoPopupHandler.cs:
```diff
@@ -0,0 +1,71 @@
+
+    public void Set(MusicTrack musicTrack, JukeboxManager.MusicTrackFavoriteType favoriteType)
+    {
+        if (_buttonHandler.MusicTrack != null && _buttonHandler.MusicTrack.Id != musicTrack.Id) ClearWeblinks();
```
Eikös tämä nyt jätä putsaamatta WebLinkejä, jos MusicTrack on null? Onko tarkoitus?

BillTheBeast committed:
Attempting to fix text encoding.

srbackman committed:
Multiple minor jukebox fixes

- Jukebox preview should not end abruptly anymore when server call is made.
- Added indicator if the music volume is set to 0.
- Added miss click prevention to jukebox lists.
- Fixed clan jukebox music desyncing if playing locally same track that is in server the current track.
- Note: There is a bug regarding SFXHandler.cs that prevents some SFX playback, cause of the bug is at the moment unknown.

BillTheBeast reviewed (approved):
Löysin vielä bugin, jossa peli joskus alkaa soittamaan In Awea, jos soittolistalla ei ole biisiä ja avaat jukeboxin uudessa ikkunassa. (Eli vaihdat ikkunaan, jossa et ole vielä ollut ja sit avaat jukeboxin.)

Lisäksi kun avaa biisintieto ikkunan ekaa kertaa ikkunassa, niin WebLink painike on aina näkyvissä, vaikka sen ei tarvitsisi olla. Se painike ei kylläkään tee mitään, joten ei ole kriittinen. (Jos suljet sen ja avaat uudestaan eri biisillä niin se ei ole enää näkyvissä.)

Mutta muuten tämä näyttäisi toimivan juuri nyt ihan ok. Hieman on ehkä turhan pienet osa noista elementeistä, mutta eiköhän Helena kommentoi sitä muutenkin myöhemmin.

Hyväksyn kumminkin tässä kohtaa.
